### PR TITLE
[EOS-22168] Multipart object overwrite with Simple object (bgdelete handling)

### PR DESCRIPTION
**Pull request by SeagateChaDeepak**
Signed-off-by: Deepak Mahale <deepak.mahale@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: server/s3_put_object_action.cc [hunk 1]
```diff
@@ -684,15 +703,30 @@ void S3PutObjectAction::add_part_object_to_probable_dead_oid_list(
...
+  s3_log(S3_LOG_DEBUG, request_id, "%s Entry\n", __func__);
+  add_object_oid_to_probable_dead_oid_list();
+  s3_log(S3_LOG_DEBUG, request_id, "%s Exit\n", __func__);
+}
+
+void S3PutObjectAction::fetch_ext_object_info_failed() {
+  // Add code in derived class to better handle
+  // failure in loading extended entries.
+  s3_log(S3_LOG_DEBUG, request_id, "%s Entry\n", __func__);
+  next();
```
Comment (knrajnambiar76) [inline server/s3_put_object_action.cc:726]: send Internal  error to client
Have logging

File: server/s3_put_object_action.cc [hunk 2]
```diff
@@ -684,15 +703,30 @@ void S3PutObjectAction::add_part_object_to_probable_dead_oid_list(
...
 }
 
+void S3PutObjectAction::fetch_ext_object_info_success() {
+  s3_log(S3_LOG_DEBUG, request_id, "%s Entry\n", __func__);
+  add_object_oid_to_probable_dead_oid_list();
+  s3_log(S3_LOG_DEBUG, request_id, "%s Exit\n", __func__);
+}
+
+void S3PutObjectAction::fetch_ext_object_info_failed() {
+  // Add code in derived class to better handle
```
Comment (knrajnambiar76) [inline server/s3_put_object_action.cc:723]: You can remove that comment

**Comment by SeagateChaDeepak** [comment]
http://eos-jenkins.colo.seagate.com/job/S3server/job/S3Server-PreMerge-Test/180/

File: server/s3_put_object_action.cc (hunk 1 above)
Comment (SeagateChaDeepak) [reply server/s3_put_object_action.cc]: Done

**Review by knrajnambiar76 — APPROVED** [review]

**Pull request merged**